sql/core/src/test/scala/org/apache/spark/sql/DateFunctionsSuite.scala

```diff
@@ -239,6 +239,21 @@ class DateFunctionsSuite extends QueryTest with SharedSQLContext {
       df.select(date_add(col("ss"), 7)),
       Seq(Row(Date.valueOf("2015-06-08")), Row(Date.valueOf("2015-06-09"))))
 
+    val df2 = df.withColumn("x", lit(1))
+    checkAnswer(
+      df2.select(date_add(col("d"), col("x"))),
```

Comment:
@WeichenXu123, I think we only need one test that calls `Column` since that's what this Pr changes.